rust/private/utils.bzl

```diff
@@ -97,17 +118,22 @@ def determine_output_hash(crate_root):
     """
     return repr(hash(crate_root.path))
 
-def get_libs_for_static_executable(dep):
+def get_libs_for_static_executable(dep, cc_toolchain, feature_configuration):
     """find the libraries used for linking a static executable.
 
     Args:
         dep (Target): A cc_library target.
+        cc_toolchain (CcToolchainInfo): The current `cc_toolchain`.
+        feature_configuration (FeatureConfiguration): The current `feature_configuration` to use with `cc_toolchain`.
 
     Returns:
         depset: A depset[File]
     """
     linker_inputs = dep[CcInfo].linking_context.linker_inputs.to_list()
-    return depset([_get_preferred_artifact(lib) for li in linker_inputs for lib in li.libraries])
+    return depset(
+        [_get_preferred_artifact(lib) for li in linker_inputs for lib in li.libraries],
+        transitive = [cc_toolchain.static_runtime_lib(feature_configuration = feature_configuration)],
```

Comment:
The [docs](https://docs.bazel.build/versions/master/skylark/lib/CcToolchainInfo.html) suggest:
>The caller should check whether the feature_configuration enables `static_link_cpp_runtimes` feature

Is there a reason to not do that here?

Reply:
Sounds good

Reply:
Last thought, sorry for missing it earlier.

get_libs_for_static_executable is called for every cc_library target.  This would result in the static_runtime_lib being depended upon N times for a target with N cc_library deps.  Is that what we want?

Reply:
Good catch! Actually the static runtime lib is in action inputs already because sad reasons, this PR only deals with passing a command line option. But you correctly notices that we put the option there multiple times, and because it's a  depset, only the first occurence will win. That will cause linking to fail in builds with multiple native libraries. I'll update the PR.

Reply:
I've uploaded a fixed version, PTAL at `add_native_link_flags`.

I'm not sure I see a way to write a test for this without having to create a C++ toolchain with static runtime lib. But I have verified it internally in our codebase. WDYT?